File: 03.Shared Preferences/MySharedPreference/app/src/main/java/com/dicoding/picodiploma/mysharedpreference/UserPreference.java
package com.dicoding.picodiploma.mysharedpreference;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Created by sidiqpermana on 11/17/16.
 */

class UserPreference {
    private static final String NAME = "name";
    private static final String EMAIL = "email";
    private static final String AGE = "age";
    private static final String PHONE_NUMBER = "phone";
    private static final String LOVE_MU = "islove";

    private final SharedPreferences preferences;

    UserPreference(Context context) {
        String PREFS_NAME = "UserPref";
        preferences = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
    }


    public void setUser(UserModel value) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putString(NAME, value.name);
        editor.putString(EMAIL, value.email);
        editor.putInt(AGE, value.age);
        editor.putString(PHONE_NUMBER, value.phoneNumber);
        editor.putBoolean(LOVE_MU, value.isLove);
        editor.apply();
    }

    void getUser(UserModel value) {
        value.setName(preferences.getString(NAME, ""));
        value.setEmail(preferences.getString(EMAIL, ""));
        value.setAge(preferences.getInt(AGE, 0));
        value.setPhoneNumber(preferences.getString(PHONE_NUMBER, ""));
        value.setLove(preferences.getBoolean(LOVE_MU, true));
    }
}

File: Shared Preferences/MySharedPreference/kotlin/app/src/main/java/com/dicoding/picodiploma/mysharedpreference/UserPreference.kt
package com.dicoding.picodiploma.mysharedpreference

import android.content.Context
import android.content.SharedPreferences
import androidx.core.content.edit

/**
 * Created by sidiqpermana on 11/17/16.
 */

internal class UserPreference(context: Context) {

    private val preferences: SharedPreferences

    var user: UserModel
        get() {
            val model = UserModel()
            model.name = preferences.getString(NAME, "")
            model.email = preferences.getString(EMAIL, "")
            model.age = preferences.getInt(AGE, 0)
            model.phoneNumber = preferences.getString(PHONE_NUMBER, "")
            model.isLove = preferences.getBoolean(LOVE_MU, false)

            return model
        }
        set(value) {
            preferences.edit {
                putString(NAME, value.name)
                putString(EMAIL, value.email)
                putInt(AGE, value.age)
                putString(PHONE_NUMBER, value.phoneNumber)
                putBoolean(LOVE_MU, value.isLove)
            }
        }

    init {
        preferences = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
    }

    companion object {
        private const val PREFS_NAME = "user_pref"
        private const val NAME = "name"
        private const val EMAIL = "email"
        private const val AGE = "age"
        private const val PHONE_NUMBER = "phone"
        private const val LOVE_MU = "islove"
    }
}
